### PR TITLE
Create a `quasar.contrib.matryoshka` package.

### PR DESCRIPTION
**Pull request by sellout**
Move the scattered things that should be pushed up to Matryoshka here.

### CODE REVIEW

File: connector/src/main/scala/quasar/qscript/ProjectBucket.scala [hunk 1]
```diff
@@ -33,6 +34,8 @@ import scalaz._, Scalaz._
   */
 sealed abstract class ProjectBucket[T[_[_]], A] {
   def src: A
+  def value: FreeMap[T]
+  def name: FreeMap[T]
```
Comment (mossprescott) [inline connector/src/main/scala/quasar/qscript/ProjectBucket.scala]: Seems unrelated.

**Comment by mossprescott** [comment]
Other than wondering if you meant to include the changes in `ProjectBucket`, this seems solid and a nice cleanup.

Build failure is another odd Travis timeout.

File: connector/src/main/scala/quasar/qscript/ProjectBucket.scala (hunk 1 above)
Comment (sellout) [inline connector/src/main/scala/quasar/qscript/ProjectBucket.scala]: Yep, good catch. That’s something from the work I extracted this PR from.

**Comment by sellout** [comment]
Ok, this is updated. Hopefully Travis behaves.

**Comment by alissapajer** [comment]
@sellout merge conflict now

**Comment by alissapajer** [comment]
@sellout btw I like this a lot. Then we can make easy dumps from quasar to matryoshka, as long as we put everything that should be in matryoshka in the contrib section :)

**Comment by mossprescott** [comment]
Timed out. Restarted.

**Comment by alissapajer** [comment]
merged after go-ahead from @mossprescott
